### DISCUSSION TITLE
Trying to get property 'name' of non-object

### DISCUSSION THREAD

**Discussion by nick-friedrich** [osiset/laravel-shopify#991] [Q&A] [answered]
Hello,

maybe you can help me out. I am trying to authenticate my laravel app with Shopify but it won't work. 

This is the error I receive when calling /authenticate/token:
`ErrorException
Trying to get property 'name' of non-object (View: /Users/nickfriedrich/www/app01/vendor/osiset/laravel-shopify/src/resources/views/layouts/default.blade.php)
https://app01.test/authenticate/token `

If I only call /authenticate I get this error:
`Error
Call to a member function getDomain() on null
https://app01.test/authenticate/ `

I am using Laravel Valet with SSL enabled. 
I used this to get the Nginx server working, before I only got a blank screen because you can't remove this line because it doesn't exist in Valet's Nginx config: `add_header X-Frame-Options "SAMEORIGIN";`

https://gist.github.com/poul-kg/b669a76fc27afcc31012aa0b0e34f738

I have no clue, why this isn't working right now, because I did everything in the tutorial.

routes/web.php:
`
Route::get('/', function () {
    return view('welcome');
})->middleware(['verify.shopify'])->name('home');`

VerifyCsrfToken.php:
`class VerifyCsrfToken extends Middleware
{
    /**
     * The URIs that should be excluded from CSRF verification.
     *
     * @var array
     */
    protected $except = [
        '*',
    ];
}`

**Comment by elalaouidev** [comment]
Do you have a paid ngrok subscription ? because I see that you are trying to access the shopify app locally, I think it wont work because your app bridge may be set to true, any way , try to use the ngrok url instead if you are using a free subscription
here's an example of a working test I did with arch + sqlite (start from 7:52) https://www.youtube.com/watch?v=5ytuGZ6Fdkw&t=141s

**Reply by nick-friedrich** [reply]
According to the wiki this shouldn't be an issue: https://github.com/osiset/laravel-shopify/wiki/Developing-Locally

**Comment by nick-friedrich** [comment]
I think I found the issue here: https://github.com/osiset/laravel-shopify/issues/912

I also tried to directly call the root / from my browser. I have no idea how I can access this page through Shopify admin dashboard.

**Comment by nick-friedrich** [comment, marked as answer]
I finally found it. Everything was correct so far. I thought I could go directly to the browser and install the app instead I had to go through Shopify partner dashboard and click on install. I am a noob! After this, I had an issue with "There's no page at this address". All I had to do, was to remove the shop user in the database and try again. Now it worked and the app is installed! Hope I can help somebody in the future with my thread.

This helped me out with my issues: https://github.com/osiset/laravel-shopify/issues/912

**Reply by elalaouidev** [reply]
Good for you =)
